### ISSUE TITLE
bug (YouTube Music/SponsorBlock): Tapping next on the media panel skips multiple songs when the next song has SponsorBlock segment at the beginning

### ISSUE THREAD

**Issue by Lassie111** [inotia00/ReVanced_Extended#2379]
### Type

Error at runtime

### Tools used

CLI

### Application

YouTube Music 7.17.51

### Bug description

1. Find a song that has SponsorBlock segment at the beginning (e.g. [](https://music.youtube.com/watch?v=RnBT9uUYb1w)).
2. Go to the previous song.
3. Minimize YouTube Music to the background.
4. From the Media panel, tap the next button.
5. Observe the whole next song being skipped (and sometimes even a few more songs after it).

Don't forget step 3! Skipping from the Media panel works fine if YouTube Music is still on the foreground.

### Device Environment

Android 14, One UI 6.1

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I didn't use any settings marked as `Experimental Flags`.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by Lassie111** [comment]
I can confirm it's fixed in the `dev` branch.

**Comment by inotia00** [comment]
I am not closing this issue right now because it has not been merged into the main branch yet
(most users do not use the dev branch)

Once it is merged into the main branch, the issue will be closed

**Comment by Lassie111** [comment]
However, I see that now when going to the previous song and that previous song has SB segment at the beginning and YT music is in the background, this segment is not skipped. I works fine if YT music is in the foreground.

**Comment by inotia00** [comment]
Probably the issue has been around for a long time

It doesn't seem easy to fix, but it's better to open a new issue for anyone who comes here to report the same issue

**Comment by inotia00** [comment]
reflected in [revanced-patches-4.14.1](https://github.com/inotia00/revanced-patches/releases/tag/v4.14.1)
